### ISSUE TITLE
Getting attention maps

### ISSUE THREAD

**Issue by debarshigit** [btzyd/F3#3]
Hi,

Congratulations for the great work.

I have a doubt regarding getting the attention maps in the LLaVA.py file.

attention = torch.cat(outputs.attentions[0], dim=0)[:,:,-1,35:35+576]
return attention

The outputs.attentions is giving a tuple of length 2, where the first element is the prefill attention and the second element is the decoded attention. Is outputs.attentions[0] taking the last row of the prefill attention, which is the attention of the last query token with the image patch tokens? But the paper denotes the cross-modal attention as the attention between the first response token and the image patch tokens. Is this somehow the same or I am missing something. It would be extremely helpful if you could clarify this part a bit.

Looking forward for your response. 
Thank you.

**Comment by btzyd** [comment]
Hi there,

So sorry for my late reply—I didn’t receive any email notifications for this issue and missed your message!

To clarify your question: In the code, we extract the attention from the last query token of the prefill phase. The first response token in the decode phase (as mentioned in the paper) is also a valid choice, and we actually tested both approaches in our experiments.

We apologize for the inconsistency between the code implementation and the paper description, which caused confusion for you.

Feel free to let me know if you have any further questions or want to discuss more details—I’m happy to help!

Best regards,
Yudong.
